controlsfx/src/main/java/org/controlsfx/control/TaskProgressView.java

```diff
@@ -62,20 +62,24 @@
  */
 public class TaskProgressView<T extends Task<?>> extends ControlsFXControl {
 
+    private boolean retainTaskMode = false;
```

Comment:
On controls we use properties, not fields. Please use BooleanProperty retainTasks = new SimpleBooleanProperty(this, "retainTasks", false); Also no reason to call it a "mode". A simple "retainTasks" is all we need.

Reply:
Also, now that the tasks are not removed automatically upon completion it would be good to have a "clear" button to manually remove them. I just noticed that the skin of TaskProgressView already creates a BorderPane but does not use it. In the end the skin adds the ListView directly to the control, the BorderPane does not get used. I would suggest you set the ListView in the center of the BorderPane and place a button at the bottom of the list view. Only show the button if the retainTasks property is "true". This can be done by binding the "visible" and the "managed" property of the clear button to the "retainTasks" property. This also shows you how important it is to use a property for "retainTasks" instead of just a field.

Reply:
I checked and I think the CSS styling will still work even when adding the BorderPane.

Reply:
hi @dlemmermann just committed the refactored code to use properties in 9cfbe5ee29e194e8c16b88a7334936150a2aee72. Honestly I don't think I have the capability to "properly" add the additional button, so if ok for you I'll keep it down to this!